sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/Configs.java

```diff
@@ -20,6 +20,18 @@
 
 public class Configs {
     private static final Logger logger = LoggerFactory.getLogger(Configs.class);
+    public static final String SPECULATION_ENABLED = "COSMOS_SPECULATION_ENABLED";
+
+    /**
+     * Integer value specifying the speculation type
+     * <pre>
+     * 0 - No speculation
+     * 1 - Threshold based speculation
+     * 2 - Thompson Sampling based speculation
+     * </pre>
+     */
+    public static final String SPECULATION_TYPE = "COSMOS_SPECULATION_TYPE";
```

Comment:
I am wondering whether we should provide both choices (threshold based and Thompson sampling). each choices have its strong and weakness, when we try to improve those areas, eventually it may drive into one hybrid solution. 
Should we start with one choice and keep making improvement on?

Reply:
And another question: should we give choice 0

Reply:
On 0 - no speculation, that is just for sanity. Any value other than 1 or 2 will essentially mean no speculation.

On the first question, yes until we get to an optimal implementation we can test both and see how that goes